src/Microsoft.SqlTools.ServiceLayer/Agent/Contracts/AgentJobHistoryInfo.cs

```diff
@@ -58,4 +58,14 @@ public  class AgentJobStep
         public CompletionResult runStatus;
         public AgentJobStepInfo stepDetails;
 	}
+
+    /// <summary>
+    /// a class for storing various properties of a agent notebook history
+    /// </summary>
+    public class AgentNotebookHistoryInfo : AgentJobHistoryInfo
+    {
+        public int MaterializedNotebookId { get; set; }
+        public int MaterializedNotebookErrorFlag { get; set; }
+        public string MaterializedNotebookErrorInfo { get; set; }
```

Comment:
check other classes for naming conventions, but I think this would be something like `ErrorMessage`